Improve start-nat-node.sh - make sure container is stopped

NumberFour8 opened the pull request:
Fixes `start-nat-node.sh` script: assign a name to the NAT'd container and make sure it is stopped.

Refs #3296

tolbrino commented on scripts/nat/start-nat-node.sh:
```diff
@@ -25,7 +25,9 @@ fi
 declare healthcheck_port=${HOPRD_HEALTHCHECK_PORT:-8080}
+
 declare network_name="hopr-nat"
+declare container_name="hoprd-behind-nat"
```
This will prevent the use of this script multiple times on the same machine. Can you parameterize this?

NumberFour8 replied:
Added env variable for this.

tolbrino commented on scripts/nat/start-nat-node.sh:
```diff
@@ -35,8 +37,12 @@ if [ "$(docker network ls | grep -c "${network_name}" )" = "0" ] ; then
+
 # Fork here and pass all the environment variables down into the forked image
 docker run --pull always -v /var/hoprd/:/app/db -p ${admin_port}:3000 -p ${rest_port}:3001 -p ${healthcheck_port}:8080 \
+ --name=${container_name} --rm \
```
This should actually also use `--restart=unless-stopped`.

robertkiel reviewed (approved):
lgtm

tolbrino commented:
@NumberFour8 You will also need to adapt the restart policy in the `scripts/gcloud.sh` helper function. Currently its `always`, meaning the top-level Docker container will run again and again, even though it exits with code 0.

See https://docs.docker.com/engine/reference/run/#restart-policies---restart

`no` or `onFailure` might be better.

NumberFour8 commented:
> on-failure

Good point, added `on-failure` as I believe that makes the most sense.

> 

I updated the `restart-policy` as follows:
- `gcloud.sh` will use `restart-policy=on-failure` for all container VMs
- the NAT nodes will be started as daemonized, allowing the bootstrapping container to exit gracefully
- `restart-policy=on-failure` is also used to start the NAT node.
